### ISSUE TITLE
Reproducing the results

### ISSUE THREAD

**Issue by trevor-taeyeop** [zju3dv/pvnet#91]
Hi, pengsida

Thank you for sharing this great work!
I have a simple question about the training approach. 

 **When I reproducing the results, the results could be many effected by the dataset which I just doing the same method in pvnet-rendering.**    #78 

I tried reproducing the results and fine-tuning using the pre-trained weights. But the results are the same. There is still gap between my own training(reproducing,finetune) and pretrained results or paper results.

- Therefore, I want to try in the same environment in your dataset which is exactly same. Is it possible?? 

- And Do you have some issues about the training dataset harm the result even though the architecture was the same.  I wonder your opinion.

For example, I tested the cat in occlusion linemod
> pretrained cat (199 epoch)   [projection 0.6276 / add 0.1887 / cm 0.1920]
> my train cat     (199 epoch)   [projection 0.5324 / add 0.1575 / cm 0.1145]

**Comment by pengsida** [comment]
What is your batch size?
Previously I uploaded rendering images of the cat model at https://github.com/zju3dv/pvnet/issues/19
It was generated by pvnet-rendering, too.

**Comment by trevor-taeyeop** [comment]
I tried batch size 32 and 16. 
Still, there is no difference in performance.
Is it possible also fusion file ??

**Comment by pengsida** [comment]
I am not sure.

**Comment by ShougangShen** [comment]
@pengsida Hi,
could you please tell me what the test result meant?  "val epoch 200 step 200 seg 0.00221293  ver 0.00179207  precision 0.95590049  recall 0.95489734  projection_error 0.99700599  add 0.77245509  cm 0.98303393 epoch 200 val cost 145.2954065799713 s"
what does **seg**, **ver** or **cm** mean?
Thanks!

**Comment by pengsida** [comment]
seg and ver mean losses for segmentation and vector-field prediction.
cm is a metric for 6d pose estimation.

**Comment by ShougangShen** [comment]
Thanks for your reply.

**Comment by trevor-taeyeop** [comment]
- I still have low 2d projection metric result than pretrained.
#57 In this result, I also close result In LINEMOD Dataset compared to pretrained weights.
But In Occluded Linemod only projection metric is lower about 10% than pretrained one.  

- If anyone has ever reproducing and evaluate in occluded linemod, can you share the results?
I want to know other reproducing results.
I think the result difference comes to the accuracy of vertex. But I don't know why the difference is  happend.

**Comment by AingLee** [comment]
> * I still have low 2d projection metric result than pretrained.
>   #57 In this result, I also close result In LINEMOD Dataset compared to pretrained weights.
>   But In Occluded Linemod only projection metric is lower about 10% than pretrained one.
> * If anyone has ever reproducing and evaluate in occluded linemod, can you share the results?
>   I want to know other reproducing results.
>   I think the result difference comes to the accuracy of vertex. But I don't know why the difference is  happend.

occ_val epoch 200 step 200 seg 0.00420556  ver 0.02362387  precision 0.85987866  recall 0.78237575  projection_error 0.63352991  add 0.19208088  cm 0.19292334 
epoch 200 occ_val cost 64.04526495933533 s

@trevor-taeyeop , is it the same as yours?

**Comment by YaohangHan** [comment]
> What is your batch size?

What is your batch size on the training approach?

**Comment by pengsida** [comment]
https://github.com/zju3dv/pvnet/blob/master/configs/linemod_train.json#L9

**Comment by XinYuANU** [comment]
Hi Pengsida,

Thanks for the great work! However, I also met the same problem using the same protocols I cannot reproduce the results, which is worse than your reported ones. 

I also noticed that you have released the data(rendered cat) and all the pre-trained models on Linemod. Therefore, I directly test the pre-trained models on the test dataset, and get the similar results are reported in the paper. But there is two strange phenomena I hope you can explain to me please:
I use your pre-trained model "cat" and then test it on the real training and fused real training data, the vertex losses are very small, around 0.002~0.001. However, I directly use your pre-trained model to evaluate on the rendered cat images as released by you. I found the errors are very large, around 0.1~0.2
This would imply the rendered cat images are not used for training? 

Moreover, I also use the real-train and fused images (without using the rendered data) to fine-tune your pre-trained model, with learning rate 1e-5. However, the performance on the test dataset drops after 1 or 2 epochs. I wonder if the model converges around 200 epochs, why the performance starts dropping by adding another one epoch? I am really confused about these two phenomena. 

Could you please just double-check with these strange phenomena? 

Thank you very much!
Best luck for the future research!

**Comment by pengsida** [comment]
Thanks for your careful examination.

1. We used the rendered cat images for training. I did not observe this phenomenon before. Maybe there are some bugs within the rendering pipeline.
2. The second phenomenon is normal for the network training. The performance vibrates during training. If you train the network more epochs, you probably get a higher performance.

I will check the reproducibility in two months.

**Comment by XinYuANU** [comment]
Thank you very much, Pensida.
For the first one, I even draw the feature points on the rendered images, and they look fine. I think rendering is fine.
For the second one, I keep training using the rendered data and fused and real images, the test performance actually drops. Please have a look.

Thank you very much for your reply. Much appreciate it!

**Comment by zeal-github** [comment]
@pengsida Hi, did you use the fuse dataset for training? Since I cannot find any statement in the readme, so I want to confirm this. Also, when I reproducing the result, it seems that without the fuse dataset(only real and render set), the results are more close to the reported one.

**Comment by pengsida** [comment]
@zeal-github We did use the fused images.
We have been refactoring the code recently and checking the producibility.

**Comment by XinYuANU** [comment]
Thank you very much for the refactoration. May I also ask how do you train the dataset on YCB? I found the syn dataset is not accurate and the dataset is not available anymore. Moreover, because there are 21 objects, may I ask how do you use fuse images for training? Thanks very much!

**Comment by pengsida** [comment]
1. https://github.com/zju3dv/pvnet/issues/4
2. What do you mean by "the syn dataset is not accurate and the dataset is not available anymore"
3. Fusing ycb is similar to fusing linemod.

**Comment by XinYuANU** [comment]
Thank you for your reply! In their original YCB dataset provided by the authors there was a folder named "synthetic". I am not sure whether you also used that data for training or not? 
YCB dataset cannot be downloaded anymore due to the large amount of downloading...
Since previously I found the synthetic data were not accurate, I deleted the folder oops... I wonder what the kinds of data you used for training on YCB? such as, the videos from 0~47 as well as rendered (10K) and fused (10K) images similar to Linemod? In that case, I cannot achieve the same performance as reported in the suppl, material. I probably miss some things. Could you please point it out? 
Many thanks!

**Comment by pengsida** [comment]
1. We used the `data_syn` under ycb dataset for training.
2. My partner trained this dataset, so I am not sure if he used rendered and fused images.

**Comment by XinYuANU** [comment]
Thank you very much for your info. Really appreciate your help!
Cheers!

**Comment by pengsida** [comment]
@XinYuANU We re-trained the network and got the following curve
![image](https://user-images.githubusercontent.com/11582122/70423521-9d0e2800-1aa8-11ea-89a2-8169624fb59d.png)
